crypto/openssl/crypto/bn/bn_const.c

```diff
@@ -82,10 +82,12 @@ BIGNUM *BN_get_rfc2409_prime_1024(BIGNUM *bn)
  * RFC2312 specifies a generator of 22.
  */
 
+#ifndef FIPS_MODULE
```

Comment:
Why exactly is the ifdef needed? It's not mentioned in the commit log.

Reply:
I mentioned it in the pull request here:

> It is not exactly clear to me why this required the change in `crypto/openssl/crypto/bn/bn_const.c`, however without it the FIPS module fails with:
> ```
> 00B0AD22712F0000:error:12800067:DSO support routines:dlfcn_load:could not load the shared library:/usr/src/crypto/openssl/crypto/dso/dso_dlfcn.c:118:filename(/usr/lib/ossl-modules/fips.so): /usr/lib/ossl-modules/fips.so: Undefined symbol "ossl_bignum_modp_1536_p"
> ```

For some reason it builds fine in the `security/openssl30` port, without this patch. I tried to compare compiler flags, but did not see what could influence that.

Reply:
Sorry I missed that.

The same ifdef wraps `modp_1536_p` in bn_dh.c, so this makes some sense. Still doesn't explain why upstream openssl doesn't have this problem.

Could you please amend the commit log message to explain this a bit further?

Reply:
Done! (in bb47d4b4c79a35d4ba477a8979c42e6d72b73dd9)